File: solutions/c/hamming/1/hamming.c
#include "hamming.h"

int compute(const char *lhs, const char *rhs){
    if (strlen(lhs)!=strlen(rhs)){
        return -1;
    }
    int cnt = 0;
    while (*lhs!='\0'){
        if (*lhs!=*rhs){
            cnt++;
        }
        lhs++;
        rhs++;
    }
    return cnt;
}

File: solutions/go/hamming/1/hamming.go
package hamming

import (
    "errors"
)

func Distance(a, b string) (int, error) {
	if len(a)!=len(b){
        return 0,errors.New("Length not matching")
    }
    var cnt int = 0
    for i,_ := range a {
        if a[i]!=b[i]{
            cnt++
        }
    }
    return cnt,nil
}
